Calculate height for marker augmenatation

hunminkim98 opened the pull request:
Hello, Dr. David,

Here are some modifications related to marker augmentation. All of these changes were implemented using your existing excellent functions!

- I added a few lines to utilize the compute_height function in markerAugmentation.py.
- I changed the default value of close_to_zero_speed from 50 to 0.2. I believe the value 50 is specific to Sports2D.
- I added some code to handle the case where the mid-hip marker is missing. I also attempted to simplify the implementation as per your suggestion, but this resulted in an error because the structure stores X, Y, and Z coordinates consecutively. Processing them all at once causes issues with the sorting and indexing of the DataFrame.

And I have commented out the part in the current config that loads subject_height. If you prefer complete automation without user input, I will remove it. On the other hand, if you would like automatic height calculation to occur only when set to 'auto,' I can make some adjustments accordingly.

Additionally, parameters such as close_to_zero_speed_m, large_hip_knee_angle, and trimmed_extrema_percent can be loaded from the config to ensure that the default values are not the only ones being used.

Thank you!

hunminkim98 commented:
Oh no, it seems there's an issue with pytest. This error occurred when all frames were filtered. I'll investigate why all the values were excluded and make the necessary corrections.
_Could not compute height from Demo_MultiPerson_P3_0-100_filt_butterworth.trc. Error: At least one of the following markers is missing for computing the height of the person: RHeel, RAnkle, RKnee, RHip, RShoulder, LHeel, LAnkle, LKnee, LHip, LShoulder._

davidpagnon reviewed (commented):
Hi, 

- Great stuff!
- It does not matter what the default value is, but it is probably more important to call the function with the right values. And I'd rather have the exact same functions on both toolboxes. So L133, you should specify the right values instead of taking the default ones. Instead of `height = compute_height(Q_coords_filtered, filtered_markers)`, 
  it should be
  `height = compute_height(Q_coords_filtered, filtered_markers,
                                                                  fastest_frames_to_remove_percent=fastest_frames_to_remove_percent, close_to_zero_speed=close_to_zero_speed, large_hip_knee_angles=large_hip_knee_angles, trimmed_extrema_percent=trimmed_extrema_percent)`
- Okay! I suggested something different in the last commit for best_coords_for_measurements, could you check if it works?
I am not sure lines 115 to 131 and needed in markerAugmentation, can you try without them?

It would be nice to have the option of setting the height automatically or not indeed! And I think that heights and weights should now be moved to the [markerAugmentation] section in Config.toml.




When everything works, it would be good to remove all the prints!

davidpagnon commented:
Sorry I commented too fast on my last comment!

davidpagnon commented:
It seems like it passes the tests now, congrats! 🎉  Does it seem like it works as expected on your end?

A few minor remarks:
- Could you make participant_height = 'auto' the default behavior? 
  Also, don't forget to add it to the Config files in the subfolders for Demo_batch :)
- Do you think it is possible to let fastest_frames_to_remove_percent and other measurement parameters in the [kinematics] section instead of [project]?
- I feel like read_trc, compute_height, and best_coords_for_measurements should be moved to common.py, what do you think?
- Instead of `print(f"Subject height for {os.path.basename(trc_file)}: {height} m")`, can you do `logging.info(f"Subject height automatically calculated for {os.path.basename(trc_file)}: {height} m")`

hunminkim98 commented:
Hello, Dr. David,

I apologize for the delayed response. I wanted to reply after completing the revisions, but some unexpected tasks came up, and I could only respond now.
The feedback from your previous review has been successfully incorporated, and I also made a few minor additional adjustments to ensure everything works as expected.

From my tests, everything functioned as we discussed. Below are my responses to your latest review and suggestions:
- Yes, absolutely! Setting 'auto' as the default makes perfect sense to me.
- I had moved fastest_frames_to_remove_percent and other parameters from the [kinematics] section to the [markerAugmentation] section earlier. Are you suggesting moving them back to the [kinematics] section?
- I completely agree! Moving them to common.py for broader usage sounds like a great idea.
- Absolutely, thank you for the excellent suggestion. I will update it accordingly!
- Yes, it achieves the same purpose perfectly. In my tests, there was an issue with generating six columns, so I fixed it in the current version.

hunminkim98 commented:
I understand! Thank you for reviewing it again. I'll apply all the other suggestions!

hunminkim98 commented:
I was wondering, do you think the 'mean_angles' function could also be moved into common.py?

hunminkim98 commented:
Hello, now everything works fine as I expected.
